test/unit/snappyex_test.exs

```diff
@@ -0,0 +1,108 @@
+defmodule Unit.SnappyExTest do
+  use ExUnit.Case
+  use ExUnitProperties
+  doctest SnappyEx
+
+  @empty_stream <<0xFF, 6::little-size(24)>> <> "sNaPpY"
+
+  def assert_snappy_decompress(compressed, uncompressed) do
+    assert compressed |> Base.decode16!() |> SnappyEx.decompress() == {:ok, uncompressed}
+  end
+
+  test "empty stream w/o stream identifier" do
+    assert {:error, _} = SnappyEx.decompress("")
+  end
+
+  test "empty stream w/ stream identifier" do
+    assert {:ok, ""} = SnappyEx.decompress(@empty_stream)
+  end
+
+  test "uncompressed stream" do
+    data = "some uncompressed data"
+    checksum = <<SnappyEx.compute_checksum(data)::little-size(32)>>
+
+    size = byte_size(data) + byte_size(checksum)
+
+    header = <<0x01, size::little-size(24)>>
+    stream = Enum.join([@empty_stream, header, checksum, data])
+
+    assert SnappyEx.decompress(stream) == {:ok, data}
+  end
+
+  test "compressed stream" do
+    data = "some compressed data"
+    checksum = <<SnappyEx.compute_checksum(data)::little-size(32)>>
+
+    {:ok, compressed_data} = :snappyer.compress(data)
+    size = byte_size(compressed_data) + byte_size(checksum)
+
+    header = <<0x00, size::little-size(24)>>
+    stream = Enum.join([@empty_stream, header, checksum, compressed_data])
+
+    assert SnappyEx.decompress(stream) == {:ok, data}
+  end
+
+  test "decompress GetMetadata responses" do
+    # Uncompressed chunks
+    msg = "0011FF060000734E6150705901150000F1D17CFF0008000000000000FFFFFFFFFFFFFFFF0F"
+    # status <> length <> ...
+    "00" <> "11" <> compressed_payload = msg
```

Comment:
Why do we have "0011" in the message if we remove it afterwards?

Reply:
The test was copied from the first snappy tests we had. This was intended as an example of manually decoding a message from a peer. Do you think we should move it somewhere else? Or should I just remove it?

Reply:
I just removed it. We can get new samples from peers anyways.